instrumentation-docs/readme.md

```diff
@@ -62,12 +73,25 @@ As of now, the following fields are supported:
 
 ```yaml
 description: "Description of what the instrumentation does."
+
+# used to mark modules that do not instrument traditional libraries (e.g. methods, annotations)
+# defaults to true
+isLibraryInstrumentation: false
```

Comment:
i struggled a bit with the name for this flag, and I wonder if having "library" in the name might be confusing when we already have the notion of library vs javaagent instrumentation types. I didn't want to just exclude them outright because I would eventually like to handle these types of instrumentations in a different manner, so instead I want to classify them in some way. 

I am open to suggestions otherwise I can continue brainstorming and followup with something a little less ambiguous